cloud/docker/docker.py

```diff
@@ -1566,6 +1596,7 @@ def create_containers(self, count=1):
         else:
             params['host_config']['Memory'] = mem_limit
 
+        params['host_config']['ulimits'] = self.ulimits
```

Comment:
`ulimit` support was added to [docker-py in 1.2.0](https://github.com/docker/docker-py/blob/60d814d742067f3e349119809935f8c0cb6c3b3b/docs/change_log.md#120) and [docker API version 1.18](https://docs.docker.com/engine/reference/api/docker_remote_api/#v1-18-api-changes), so we should gate ulimit support on an `ensure_capabilities` call. You can do this by adding an entry to [`_cap_ver_req`](https://github.com/dusdanig/ansible-modules-core/blob/devel/cloud/docker/docker.py#L654-L669) and then gating this call:

```python
if self.ulimits is not None:
  self.ensure_capability('ulimits')
  params['host_config']['ulimits'] = self.ulimits
```